hts-precompile/IHederaTokenService.sol

```diff
@@ -441,6 +569,189 @@ interface IHederaTokenService {
+
+    /// Operation to freeze token account
+    /// @param token The token address
+    /// @param account The account address to be frozen
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function freezeToken(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to unfreeze token account
+    /// @param token The token address
+    /// @param account The account address to be unfrozen
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function unFreezeToken(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to grant kyc to token account
+    /// @param token The token address
+    /// @param account The account address to grant kyc
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function grantTokenKyc(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to revoke kyc to token account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function revokeTokenKyc(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to pause token
+    /// @param token The token address to be paused
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function pauseToken(address token) external returns (int responseCode);
+
+    /// Operation to unpause token
+    /// @param token The token address to be unpaused
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function unPauseToken(address token) external returns (int responseCode);
+
+    /// Operation to wipe fungible tokens from account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @param amount The number of tokens to wipe
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function wipeTokenAccount(
+        address token,
+        address account,
+        uint32 amount
+    ) external returns (int responseCode, uint32 wipedAmount);
+
+    /// Operation to wipe non fungible tokens from account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @param  serialNumbers The serial numbers of token to wipe
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function wipeTokenAccount(
+        address token,
+        address account,
+        uint64[] memory serialNumbers
+    ) external returns (int responseCode, uint64[] memory wipedSerialNumbers);
```

Comment:
Same comment as the fungible wipe function --- maybe remove the `wipedSerialNumbers`?  [If a single NFT wipe fails, the whole execution fails.](https://github.com/hashgraph/hedera-services/blob/d8c782722f670a76e7b53ef5a8c64fdde5b226d8/hedera-node/src/main/java/com/hedera/services/store/models/Token.java#L305-L310)

Reply:
Updated